library/core/src/iter/adapters/step_by.rs

```diff
@@ -230,6 +360,163 @@ where
     }
 }
 
-// StepBy can only make the iterator shorter, so the len will still fit.
-#[stable(feature = "iterator_step_by", since = "1.28.0")]
-impl<I> ExactSizeIterator for StepBy<I> where I: ExactSizeIterator {}
+macro_rules! spec_int_ranges {
```

Comment:
I think it would be nice to have a comment on this describing the approach.

A first stab:
```suggestion
// For these implementations, `SpecRangeSetup` calculates the number
// of iterations that will be needed and stores that in `iter.end`.
//
// The various iterator implementations then rely on that to not need
// overflow checking, letting loops just be counted instead.
macro_rules! spec_int_ranges {
```